website/src/main.js

```diff
@@ -5,38 +5,62 @@ import "~/assets/css/main.css";
 import Vuex from "vuex";
 import DefaultLayout from "~/layouts/Default.vue";
 
-export default function(Vue, { router, head, isClient, appOptions }) {
-  // Set default layout as a global component
-  Vue.component("Layout", DefaultLayout);
-
-  Vue.use(Vuex);
-
-  appOptions.store = new Vuex.Store({
-    state: {
-      sidebarIsOpen: false,
-    },
-
-    mutations: {
-      setSidebarIsOpen(state, val) {
-        state.sidebarIsOpen = val;
-      },
-      toggleSidebarIsOpen(state) {
-        state.sidebarIsOpen = !state.sidebarIsOpen;
-      },
-    },
-  });
-
-  router.beforeEach((to, _from, next) => {
-    head.meta.push({
-      key: "og:url",
-      name: "og:url",
-      content: process.env.GRIDSOME_BASE_PATH + to.path,
+export default function (Vue, {router, head, isClient, appOptions}) {
+    // Set default layout as a global component
+    Vue.component("Layout", DefaultLayout);
+
+    Vue.use(Vuex);
+
+    appOptions.store = new Vuex.Store({
+        state: {
+            sidebarIsOpen: false,
+        },
+
+        mutations: {
+            setSidebarIsOpen(state, val) {
+                state.sidebarIsOpen = val;
+            },
+            toggleSidebarIsOpen(state) {
+                state.sidebarIsOpen = !state.sidebarIsOpen;
+            },
+        },
+    });
+
+    router.beforeEach((to, _from, next) => {
+        head.meta.push({
+            key: "og:url",
+            name: "og:url",
+            content: process.env.GRIDSOME_BASE_PATH + to.path,
+        });
+        next();
     });
-    next();
-  });
 
-  head.script.push({
-    src: "/js/asciinema-player.js",
-    body: true,
-  });
+    head.script.push({
```

Comment:
Is it possible to import the module using npm and embed it into the page during the static build to avoid loading it dynamically on each page redraw?

Reply:
Good catch.

Reply:
![image](https://user-images.githubusercontent.com/61661187/155167312-52935a79-caf7-4d5d-b209-85e41b9ef918.png)

Here is the above code which I have written which cannot be used as npm and code is already approved kindly merge this with master Thanks

Reply:
@mohsin309 Your code is fine ... we want the JS to be handled by webpack so that each page load won't have to pull resources.

Reply:
@Unix4ever found https://www.npmjs.com/package/gridsome-plugin-algolia#usage

Reply:
I would say that the main problem with your approach is that the dependencies get scattered around the code base.
If we rely on webpack to get dependencies then it is better to consistently use it.
It will be easier to update libraries that way as well.

Reply:
Shouldn't everything here and below be removed as you now load this module using webpack?

Reply:
I don't understand can you elaborate please

Reply:
If you have installed the module through webpack, why do you still need to load it manually here?
I would expect it to work without `head.link.push`, `head.script.push` and so on.

Reply:
Kindly check this now Thanks

Reply:
yeah... Looks correct now. Thank you.

Reply:
Thanks to you really learning from you I wish to be a part of your team it will be a great chance for me to learn things in depth Thanks